wewatch-client/src/components/Match/Match.js

```diff
@@ -1,44 +1,83 @@
 import React, { useState, useEffect } from "react";
+import { Redirect } from "react-router-dom";
+
 import MovieDetail from 'views/swiping/movie-detail.js';
 import { Container, Row, Col } from 'reactstrap';
-import theme from 'styles/theme'
 import StyledDiv from 'styles/styled-div';
-
 import './Match.css';
 import Layout from 'views/layout';
+import socket from 'Socket';
 
 const Match = (props) => {
     const movieData = props.location.state.matchedMovie;
+    const isHost = props.location.state.isHost;
+    const roomId = props.location.state.roomId;
+    const name = props.location.state.name;
+
+    const [tryAgain, setTryAgain] = useState(false);
+    const [tryAgainHost, setTryAgainHost] = useState(false);
+    const [newRoomId, setNewRoomId] = useState('');
+
+    const tryAgainEmit = () => {
+        socket.emit('try_again_event', {roomId: roomId})
+    }
+
+    //listen to try again event
+    useEffect(() => {
+        socket.on('tryAgainRedirectHost', () => {
+            if (isHost) {
+                setTryAgainHost(true);
+            } 
+       })
+       return () => { socket.off('tryAgainRedirectHost')};
```

Comment:
Good catch making sure that socket.off is called here! In previous sprints, this issue caused memory leaks so ensuring we have the correct lifecycle functions being called will undoubtedly save us time debugging in the future.